### ISSUE TITLE
Specify features such as tunnel in shapes.txt

### ISSUE THREAD

**Issue by miklcct** [google/transit#520]
### Describe the problem

I would like the ability to specify if a trip goes through a tunnel.

### Use cases

I want to add a preference in the journey planner for the user to avoid tunnels, and show these tunnels on the journey plan.

### Proposed solution

A new field in the `shapes.txt` to specify features en-route, such as tunnels.

**Comment by skinkie** [comment]
Since shapes.txt does consists of nodes, instead of links, addressing a section is something that should first be cleared up. Dependend on what the target is for example cartography, allowing the rendering of tunnels, roads, bridges, or just the ability to tell if the next section passes through a tunnel, it might be more relevant to specify it in stop_times.txt instead.

**Comment by skinkie** [comment]
@miklcct it certainly is in the way you described it. You described the feature as a _filter_. A similar filter exists for trips.txt to bring a bike aboard a vehicle. When you state _I want to do a geographical description, and from that description a filter can be derived_, that is a complex action, because that would required the journey planner to align shapes.txt to stop_times.txt, figure out what section a user may want to avoid.

Like I wrote before, you are suggesting to mix carthography with filters. And did not mention this in your **Use cases**.

**Comment by miklcct** [comment]
For example, the section between two stops is 15 km with 5 km in tunnels, so I want the journey planner to apply a penalty on one third of the running time for that section.

**Comment by doconnoronca** [comment]
I was just thinking a something to describe the ROW in shapes.txt would be useful to identify where a route is in mixed traffic and predictions could take current traffic congestion into account for those sections.

**Comment by skinkie** [comment]
> For example, the section between two stops is 15 km with 5 km in tunnels, so I want the journey planner to apply a penalty on one third of the running time for that section.

But would that _matter_ for someone that is afraid of tunnels? Wouldn't that just be a binary: _no tunnels_?

**Comment by miklcct** [comment]
Sorry it isn't a binary. For someone who's afraid of tunnels, 30 seconds in a tunnel v.s. 5 minutes in a tunnel is a huge difference.

**Comment by stevenmwhite** [comment]
Is the idea here that there'd be a new column on the shapes file with an enum to indicate the type of that segment, running until the next specified type on a later point?

**Comment by miklcct** [comment]
> Is the idea here that there'd be a new column on the shapes file with an enum to indicate the type of that segment, running until the next specified type on a later point?

Yes it is. Similar things can also be used to specify right of ways instead.
